### PR TITLE
[Validator] Add `Week` constraint

### LINKED ISSUE #20109: [Validator] Add `Week` constraint
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#57908
| PR author(s) | @alexandre-daubois
| Merged in    | 7.2

We created this issue to not forget to document this new feature. We would really appreciate if you can help us with this task. If you are not sure how to do it, please ask us and we will help you.

To fix this issue, please create a PR against the 7.2 branch in the [symfony-docs repository](https://github.com/symfony/symfony-docs).

Thank you! :smiley:

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Fix #20109

### CODE REVIEW

File: reference/constraints/Week.rst [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+==========  =======================================================================
+Applies to  :ref:`property or method <validation-property-target>`
+Class       :class:`Symfony\\Component\\Validator\\Constraints\\WordCount`
+Validator   :class:`Symfony\\Component\\Validator\\Constraints\\WordCountValidator`
+==========  =======================================================================
+
+Basic Usage
+-----------
+
+If you wanted to ensure that the ``startWeek`` property of a ``OnlineCourseDTO``
```
Comment (xabbuh) [inline reference/constraints/Week.rst]:
```suggestion
If you wanted to ensure that the ``startWeek`` property of an ``OnlineCourseDTO``
```

File: reference/constraints/Week.rst [hunk 2]
```diff
@@ -0,0 +1,170 @@
+Week
+====
+
+.. versionadded:: 7.2
+
+    The ``Week`` constraint was introduced in Symfony 7.2.
+
+Validates that a string (or an object implementing the ``Stringable`` PHP interface)
+matches a given week number. The week number format is defined by the ISO-8601
```
Comment (xabbuh) [inline reference/constraints/Week.rst]:
```suggestion
matches a given week number. The week number format is defined by ISO-8601
```

or

```suggestion
matches a given week number. The week number format is defined by the ISO-8601 standard
```

File: reference/constraints/Week.rst [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+and should be composed of the year and the week number, separated by a hyphen
+(e.g. ``2022-W01``).
+
+==========  =======================================================================
+Applies to  :ref:`property or method <validation-property-target>`
+Class       :class:`Symfony\\Component\\Validator\\Constraints\\WordCount`
```
Comment (xabbuh) [inline reference/constraints/Week.rst]:
```suggestion
Class       :class:`Symfony\\Component\\Validator\\Constraints\\Week`
```

File: reference/constraints/Week.rst [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+Validator   :class:`Symfony\\Component\\Validator\\Constraints\\WordCountValidator`
```
Comment (xabbuh) [inline reference/constraints/Week.rst]:
```suggestion
Validator   :class:`Symfony\\Component\\Validator\\Constraints\\WeekValidator`
```

**Comment by alexandre-daubois** [comment]
Silly copy-paste from me, sorry :facepalm: I addressed your comments

**Review by xabbuh — APPROVED** [review]

File: reference/constraints/Week.rst [hunk 5]
```diff
@@ -0,0 +1,170 @@
+Week
+====
+
+.. versionadded:: 7.2
+
+    The ``Week`` constraint was introduced in Symfony 7.2.
+
+Validates that a string (or an object implementing the ``Stringable`` PHP interface)
+matches a given week number. The week number format is defined by ISO-8601
```
Comment (94noni) [inline reference/constraints/Week.rst]: what about link to wikipedia ?

File: reference/constraints/Week.rst [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+.. configuration-block::
+
+    .. code-block:: php-attributes
+
+        // src/Entity/OnlineCourseDTO.php
+        namespace App\Entity;
+
+        use Symfony\Component\Validator\Constraints as Assert;
+
+        class OnlineCourseDTO
```
Comment (94noni) [inline reference/constraints/Week.rst]: entity and dto seems weird 
what about removing dto ?

**Review by 94noni — APPROVED** [review]

File: reference/constraints/Week.rst (hunk 5 above)
Comment (alexandre-daubois) [reply reference/constraints/Week.rst]: Here you go

File: reference/constraints/Week.rst (hunk 6 above)
Comment (alexandre-daubois) [reply reference/constraints/Week.rst]: Agree :+1:

File: reference/constraints/Week.rst [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+            public static function loadValidatorMetadata(ClassMetadata $metadata): void
+            {
+                $metadata->addPropertyConstraint('startWeek', new Assert\Week([
+                    'min' => '2022-W01',
+                    'max' => '2022-W20',
+                ]));
+            }
+        }
+
+.. tip::
```
Comment (OskarStark) [inline reference/constraints/Week.rst]:
```suggestion
.. note::
```

Comment (OskarStark) [reply reference/constraints/Week.rst]: Do we use tip anywhere else?

File: reference/constraints/Week.rst [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+
+    The constraint also checks that the given week exists in the calendar. For example,
+    ``2022-W53`` is not a valid week number because 2022 only has 52 weeks.
```
Comment (OskarStark) [inline reference/constraints/Week.rst]:
```suggestion
    ``2022-W53`` is not a valid week number for 2022, because 2022 only had 52 weeks.
```

**Review by OskarStark — CHANGES_REQUESTED** [review]
Some small comments

File: reference/constraints/Week.rst (hunk 7 above)
Comment (alexandre-daubois) [reply reference/constraints/Week.rst]: It can happen, but you're right, a note is more appropriate here

**Comment by javiereguiluz** [comment]
Alex, thanks a lot for contributing this nice constraint and the docs for it. Thanks to reviewers too. The contents were great but I made some minor tweaks while merging (https://github.com/symfony/symfony-docs/commit/3f0fdbad1f95f056c576e2d34f1576c4fe27c720). Thanks!

**Pull request merged**